File: ui-api/src/main/java/io/tipsters/uiapi/service/CountryService.java
package io.tipsters.uiapi.service;

import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import java.util.List;

import io.tipsters.uiapi.dto.CountryData;
import io.tipsters.uiapi.repository.CountryRepository;

import static java.util.stream.Collectors.toList;
import static org.springframework.data.domain.Sort.Direction.ASC;

@Service
public class CountryService {
  private final CountryRepository countryRepository;

  public CountryService(CountryRepository countryRepository) {
    this.countryRepository = countryRepository;
  }

  public List<CountryData> countriesOrderedByPriority(int limit) {
    return countryRepository.findAll(new PageRequest(0, limit, ASC, "rank")).getContent().stream()
        .map(CountryData::of)
        .collect(toList());
  }
}

File: web-server/src/main/kotlin/io/tipsters/service/CountryService.kt
package io.tipsters.service

import io.tipsters.repository.CountriesRepository
import io.tipsters.service.data.CountryData
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort.Direction.ASC
import org.springframework.stereotype.Service

@Service
class CountryService @Autowired constructor(private val countryRepository: CountriesRepository) {

    fun countriesOrderedByPriority(limit: Int): List<CountryData> =
            countryRepository
                    .findAll(PageRequest(0, limit, ASC, "rank"))
                    .content.map { country -> CountryData.of(country) }
}
